client/state/jetpack-settings/modules/reducer.js

```diff
@@ -23,24 +47,10 @@ import {
  * @param  {Object} action action
  * @return {Array}         Updated state
  */
-export const items = ( state = {}, { type, siteId, moduleSlug } ) => {
-	switch ( type ) {
-		case JETPACK_MODULE_ACTIVATE_SUCCESS:
-			return merge( {}, state, {
-				[ siteId ]: {
-					[ moduleSlug ]: {
-						active: true
-					}
-				}
-			} );
-		// return initial state when serializing/deserializing
-		case SERIALIZE:
-		case DESERIALIZE:
-			return {};
-		default:
-			return state;
-	}
-};
+export const items = createReducer( {}, {
+	[ JETPACK_MODULE_ACTIVATE_SUCCESS ]: createItemsReducer( true ),
```

Comment:
That makes sense. So let's keep that for future PRs and adapt the structure to whatever fits best. And feel free to rename the helpers to what makes sense in the next iteration.